### ISSUE TITLE
Improvements discussion

### ISSUE THREAD

**Issue by khmyznikov** [khmyznikov/ios-pwa-wrap#1]
Feel free to suggest and discuss

**Comment by khmyznikov** [comment]
For now notification permission request shows up immediately after app starts. This needs to be changed for ability to call from JS code.

**Comment by khmyznikov** [comment]
I have made a push permission request from JS code.

**Comment by khmyznikov** [comment]
Fixed errors in requesting Location or Media APIs.

**Comment by aidar87** [comment]
Hi, thanks, I like your pwa wrapper,
I added sign in with Google account, in Safari all is ok, but in wrapper I got 403 error when fetching js script.
Not sure, seems app should be added somewhere in google cloud console

**Comment by dmytro-podolianskyi-ew** [comment]
Could you tell please, what about apple social login? I get an error [core] No URL for Apple ID Authorization
Maybe do I need to add some configs or smth like that?

**Comment by khmyznikov** [comment]
> Could you tell please, what about apple social login? I get an error [core] No URL for Apple ID Authorization
Maybe do I need to add some configs or smth like that?

Do you use [this library](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple)? I don't try it actually, not sure I can help.

**Comment by aidar87** [comment]
> Could you tell please, what about apple social login? I get an error [core] No URL for Apple ID Authorization
> Maybe do I need to add some configs or smth like that?

I added Appleid domain to WKAppBoundDomains in Info file and using this guide https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple

**Comment by dmytro-podolianskyi-ew** [comment]
Thanks, it's working!

**Comment by dmytro-podolianskyi-ew** [comment]
Guys, maybe you help me again) What about google social login? When I try to login via google I stuck on the screen with loader and text "One moment please". I've added accounts.google.com to WKAppBoundDomains as the same as apple, but it's not working. Could you help please?

**Comment by khmyznikov** [comment]
> Guys, maybe you help me again) What about google social login? When I try to login via google I stuck on the screen with loader and text "One moment please". I've added accounts.google.com to WKAppBoundDomains as the same as apple, but it's not working. Could you help please?

Try to debug via XCode and Safari dev tools. Maybe there are more domains than just accounts.google.

**Comment by aidar87** [comment]
@khmyznikov maybe it is time to add corrections to the [readme](https://github.com/khmyznikov/ios-pwa-wrap#push-permission-state) for iOS 16.4 to prevent double Push from Web Push API and native Push API (for iOS devices < 16.4)

**Comment by khmyznikov** [comment]
> @khmyznikov maybe it is time to add corrections to the [readme](https://github.com/khmyznikov/ios-pwa-wrap#push-permission-state) for iOS 16.4 to prevent double Push from Web Push API and native Push API (for iOS devices < 16.4)

You have Web Push in this wrapper?

**Comment by aidar87** [comment]
> You have Web Push in this wrapper?

Yes. But currently cannot test it on iOS 16.4 beta. Seems, there will no double Push in native wrapper, web push works only if PWA added home screen

**Comment by StillOnMyWay** [comment]
> > You have Web Push in this wrapper?
> 
> Yes. But currently cannot test it on iOS 16.4 beta. Seems, there will no double Push in native wrapper, web push works only if PWA added home screen

So to be clear, the Web Push doesn't interfere with the native push implementation in this wrapper?

**Comment by khmyznikov** [comment]
> So to be clear, the Web Push doesn't interfere with the native push implementation in this wrapper?
Yes, doesn't. Web Push doesn't work in WKWebView

**Comment by fenbru** [comment]
Hi @khmyznikov, do you have any demo repo where all functionalities are used?

Thx in advance.

**Comment by khmyznikov** [comment]
@fenbru what exactly functions? For Push there is code examples in Readme

**Comment by fenbru** [comment]
Sorry, my question wasn't specific.

Can you list all the places inside the project we should change for our website, not yours? @khmyznikov

**Comment by khmyznikov** [comment]
@fenbru Settings.swift, Info.plist. Change the urls/domains

**Comment by jcmaciel** [comment]
Hey guys, thanks for this repo, it's very useful.
Let me know if I can help you ;)

**Comment by khmyznikov** [comment]
> Hey guys, thanks for this repo, it's very useful. Let me know if I can help you ;)

Any experience in Swift? 😅

**Comment by md-riaz** [comment]
Can anyone guide me how the launch process has a splash screen with loading animation with pwa?

**Comment by khmyznikov** [comment]
@md-riaz can you specify? Not sure I got the question
